packages/tc-markdown-parser/__tests__/property-value.spec.js

```diff
@@ -334,3 +348,199 @@ describe('dealing with html comments', () => {
 		});
 	});
 });
+
+describe('relationship property tests', () => {
+	describe('single relationship case - hasMany: false', () => {
+		it('can be parsed as plain string', async () => {
+			const { data, errors } = await parser.parseMarkdownString(here`
+				# name
+
+				## favourite child
+
+				example-code
+				`);
+
+			expect(errors).toHaveLength(0);
+			expect(data).toEqual({
+				name: 'name',
+				favouriteChild: {
+					code: 'example-code',
+				},
+			});
+		});
+
+		it('can be parsed as object which contains property object', async () => {
+			const { data, errors } = await parser.parseMarkdownString(here`
+				# name
+
+				## favourite child
+
+				example-code
+					someString: i like it
+			`);
+
+			expect(errors).toHaveLength(0);
+			expect(data).toEqual({
+				name: 'name',
+				favouriteChild: {
+					code: 'example-code',
+					someString: 'i like it',
+				},
+			});
+		});
+	});
+
+	describe('thorws syntax error on mutiline defitniion', () => {
+		it('mutiline property name should be a lower camel case', async () => {
+			const { data, errors } = await parser.parseMarkdownString(here`
+				# name
+
+				## favourite child
+
+				example-code
+					some string: i like it
+			`);
+
+			expect(errors).toHaveLength(1);
+			expect(data.name).toEqual('name');
+			const [{ message }] = errors;
+			expect(message).toMatch(/should be lower camel case/);
+			expect(message).toMatch(/line 6/);
```

Comment:
It was hard to overwrite `line` property in unified AST, so I intended to include the actual line number inside the error message